packages/pluggableWidgets/barcode-scanner-web/src/components/BarcodeScanner.tsx

```diff
@@ -64,35 +45,26 @@ export function BarcodeScannerOverlay({
     );
 }
 
+export interface BarcodeScannerProps extends Dimensions {
+    onDetect?: (data: string) => void;
+    showMask: boolean;
+    class: string;
+}
+
 export function BarcodeScanner({
     onDetect,
     showMask,
     class: className,
     ...dimensions
 }: BarcodeScannerProps): ReactElement | null {
-    const videoElement = useRef<HTMLVideoElement | null>(null);
-    const hasDetectedOnce = useRef<boolean>(false);
-    const { streamObject, error: errorMediaStream } = useMediaStream();
-    const { codeResult, error: errorCodeScanner } = useCodeScanner(streamObject, videoElement.current);
-    const supportsCameraAccess = browserSupportsCameraAccess();
-
-    // If both the video element ref and the camera stream object are ready, display it through the `srcObject` prop.
-    useEffect(() => {
-        if (videoElement.current && streamObject) {
-            videoElement.current.srcObject = streamObject;
+    const [errorMessage, setError] = useCustomErrorMessage();
+    const videoRef = useReader({ onSuccess: onDetect, onError: setError });
```

Comment:
I think it would better if we rename this variable, I was expecting this to be element because of the Ref, but turns out it's just a state. maybe `videoElementState`?

Reply:
but this is ref because it created with `useRef`, so I think it's better to keep this name

Reply:
eem, yah it has ref inside the method, but still the variable, the final one is not a Ref.

Reply:
I don't get it. We create ref, return ref, and passing ref to `<video />` why should we use state?

Reply:
Now it's clear!